src/Storages/ExternalStream/Kafka/Kafka.h

```diff
@@ -5,21 +5,45 @@
 #include <Storages/ExternalStream/ExternalStreamSettings.h>
 #include <Storages/ExternalStream/StorageExternalStreamImpl.h>
 #include <Storages/ExternalStream/ExternalStreamCounter.h>
+#include <Storages/ExternalStream/Kafka/Consumer.h>
+#include <Storages/ExternalStream/Kafka/ConsumerPool.h>
+#include <Storages/ExternalStream/Kafka/Producer.h>
 #include <Storages/Streaming/SeekToInfo.h>
 
 namespace DB
 {
 
 class IStorage;
 
+
 class Kafka final : public StorageExternalStreamImpl
 {
 public:
+    using ConfPtr = std::unique_ptr<rd_kafka_conf_t, decltype(rd_kafka_conf_destroy) *>;
+
+    static Poco::Logger * cbLogger() {
+        static Poco::Logger * logger { &Poco::Logger::get("KafkaExternalStream") };
+        return logger;
+    }
+
+    static int onStats(struct rd_kafka_s * rk, char * json, size_t json_len, void * opaque);
+    static void onError(struct rd_kafka_s * rk, int err, const char * reason, void * opaque);
+    static void onThrottle(struct rd_kafka_s * rk, const char * broker_name, int32_t broker_id, int throttle_time_ms, void * opaque);
+
     Kafka(IStorage * storage, std::unique_ptr<ExternalStreamSettings> settings_, const ASTs & engine_args_, bool attach, ExternalStreamCounterPtr external_stream_counter_, ContextPtr context);
     ~Kafka() override = default;
 
-    void startup() override { }
-    void shutdown() override { }
+    void startup() override { LOG_INFO(logger, "Starting Kafka External Stream"); }
+    void shutdown() override {
+        LOG_INFO(logger, "Shutting down Kafka External Stream");
```

Comment:
we can add external stream name info

Reply:
The `logger` already contains the stream name.